src/jit/morph.cpp

```diff
@@ -7689,12 +7706,36 @@ GenTreePtr Compiler::fgMorphCall(GenTreeCall* call)
                 //
                 if (call->IsImplicitTailCall())
                 {
-                    if (varDsc->lvHasLdAddrOp || varDsc->lvAddrExposed)
+                    if (varDsc->lvHasLdAddrOp)
                     {
                         hasAddrExposedVars = true;
                         break;
                     }
-                    if (varDsc->lvPromoted && varDsc->lvIsParam)
+                    else if (varDsc->lvAddrExposed)
```

Comment:
Right, I think I see that, but my point was that the previous `if` still ends in a `break;` so there's no need for the else if (just as before). However, I'm pretty sure it will make exactly zero difference in the generated code.

---
In reply to: [116548706](https://github.com/dotnet/coreclr/pull/10453#discussion_r116548706) [](ancestors = 116548706)